File: PoojaSeva/app/src/main/java/com/poojaseva/domain/model/Booking.kt
package com.poojaseva.domain.model

data class Address(
    val line1: String,
    val landmark: String?,
    val city: String,
    val state: String?,
    val pincode: String,
)

enum class BookingStatus { Pending, Confirmed, InProgress, Completed, Cancelled, Unknown }

data class Booking(
    val id: String,
    val serviceId: String,
    val serviceName: String,
    val panditName: String?,
    val scheduledAtEpochMillis: Long,
    val address: Address,
    val contactName: String,
    val contactPhone: String,
    val notes: String,
    val totalInr: Int,
    val status: BookingStatus,
    val createdAtEpochMillis: Long,
)

/** What the booking form collects before a booking exists on the server. */
data class BookingDraft(
    val serviceId: String,
    val serviceName: String,
    val scheduledAtEpochMillis: Long,
    val address: Address,
    val contactName: String,
    val contactPhone: String,
    val notes: String,
    val totalInr: Int,
)

File: PoojaSeva/backend/app/schemas/booking.py
from datetime import datetime
from pydantic import BaseModel, ConfigDict


class BookingCreate(BaseModel):
    user_id: int | None = None
    service_id: str
    service_name: str
    pandit_id: str | None = None
    pandit_name: str | None = None
    scheduled_at: datetime
    address_line: str
    landmark: str | None = None
    city: str
    state: str | None = None
    pincode: str
    contact_name: str
    contact_phone: str
    notes: str | None = None
    total_inr: int


class BookingOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    user_id: int | None = None
    service_id: str
    service_name: str
    pandit_id: str | None = None
    pandit_name: str | None = None
    scheduled_at: datetime
    address_line: str
    landmark: str | None = None
    city: str
    state: str | None = None
    pincode: str
    contact_name: str
    contact_phone: str
    notes: str | None = None
    total_inr: int
    status: str
    created_at: datetime


class BookingStatusUpdate(BaseModel):
    status: str
